src/leaflike/bookmarks/views.clj

```diff
@@ -16,16 +16,34 @@
       [:a {:class "page-link"
            :href (path-format-fn (dec current-page))}
        "Previous"]])
+
    ;; List of pages
-   (for [page-num (range 1 (inc num-pages))]
-     (let [active? (= page-num current-page)
-           li-class "page-item"
-           li-class (if active?
-                      (str li-class " active")
-                      li-class)]
-       [:li {:class li-class}
-        [:a {:class "page-link"
-             :href (path-format-fn page-num)} page-num]]))
+   (let [window-size (min 4 num-pages)
```

Comment:
@samrat This is a rather large `let` that obfuscates the idea that it's really `visible-pages` that you need for constructing the view. How about pulling that out into a separate fn that encapsulates it all and returns `visible-pages` ?

Reply:
Fixed this.